File: Problems/Codeforces/Division 2/100's/137/Cosmic Tables.py
import sys
input = sys.stdin.readline
def print(val):
    sys.stdout.write(str(val) + '\n')
n,m,k = map(int,input().split())
table = [list(input().split()) for i in range(n)]
curr_row = [i for i in range(n)]
curr_col = [i for i in range(m)]
for _ in range(k):
    q = input().split()
    q[2] = int(q[2]) - 1
    q[1] = int(q[1]) - 1
    if q[0] == 'c':
        a = curr_col[q[2]]
        curr_col[q[2]] = curr_col[q[1]]
        curr_col[q[1]] = a
    elif q[0] == 'r':
        a = curr_row[q[2]]
        curr_row[q[2]] = curr_row[q[1]]
        curr_row[q[1]] = a
    else:
        print(table[curr_row[q[1]]][curr_col[q[2]]])

    

File: Problems/Codeforces/Division 2/100's/137/Cosmic Tables.cpp
#include<bits/stdc++.h>
using namespace std;

int main() {
	ios_base::sync_with_stdio(0); cin.tie(0);
	int n, m, k;
	cin >> n >> m >> k;
	vector<vector<int>> table(n, vector<int>(m));
	for (int i = 0; i < n; i++) {
		for (int j = 0; j < m; j++) {
			cin >> table[i][j];
		}
	}
	vector<int> curr_row(n);
	for (int i = 0; i < n; i++) {
		curr_row[i] = i;
	}
	vector<int> curr_col(m);
	for (int i = 0; i < m; i++) {
		curr_col[i] = i;
	}
	while (k--) {
		char c;
		int x, y; 
		cin >> c >> x >> y;
		x--; y--;
		//cout << c << ' ' << x << ' ' << y << '\n';
		if (c == 'c') {
			swap(curr_col[x], curr_col[y]);
		}
		else if (c == 'r') {
			swap(curr_row[x], curr_row[y]);
		}
		else {
			cout << table[curr_row[x]][curr_col[y]] << '\n';
		}
	}
}